keps/sig-node/3327-align-by-socket/README.md

```diff
@@ -0,0 +1,252 @@
+  - Feature gate name: `CPUManagerPolicyAlphaOptions`
+  - Components depending on the feature gate: `kubelet`
+- [X] Change the kubelet configuration to set a `CPUManager` policy of `static` and a `CPUManager` policy option of `align-by-socket`
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+	Yes -- a kubelet restart is required.
+
+###### Does enabling the feature change any default behavior?
+
+No. In order to trigger any of the new logic, three things have to be true:
+1. The `CPUManagerPolicyOptions` feature gate must be enabled
+1. The `static` `CPUManager` policy must be selected
+1. The new `align-by-socket` policy option must be selected
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled by either:
+1. Disabling the `CPUManagerPolicyOptions` feature gate
+1. Switching the `CPUManager` policy to `none`
+1. Removing `align-by-socket` from the list of `CPUManager` policy options
+
+Existing workloads will continue to run uninterrupted, with any future workloads having their CPUs allocated according to the policy in place after the rollback.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+No changes. Existing container will not see their allocation changed. New containers will.
+
+###### Are there any tests for feature enablement/disablement?
+
+- A specific e2e test will demonstrate that the default behaviour is preserved when the feature gate is disabled, or when the feature is not used (2 separate tests)
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Inspect the kubelet configuration of a node -- check for the presence of the feature gate and usage of the new policy option.
+
+###### How can someone using this feature know that it is working for their instance?
+
+In order to verify this feature is working, one should:
+Pick a node with at least 2 Sockets and 8 NUMA nodes
+Ensure no other pods with exclusive CPUs are running on that node
+Launch a 2 pods with a nodeSelector to that node that has a single container in it
+Run a `sleep infinity` command and request exclusive CPUs for the container in the amount of (4*NUM_CPUS_PER_NUMA_NODE - 8)
+Verify that for both pods, all CPU’s are within same socket instead of cpu’s distributed across sockets
+
+To verify the list of CPUs allocated to the container, one can either:
+- `exec` into uthe container and run `taskset -cp 1` (assuming this command is available in the container).
+- Call the `GetCPUS()` method of the `CPUProvider` interface in the `kubelet`'s [podresources API](https://pkg.go.dev/k8s.io/kubernetes/pkg/kubelet/apis/podresources#CPUsProvider).
```

Comment:
I think it is sufficient to connect to the podresources API and send a `List` call; inspecting the input should provide all the needed informations.
I know of a few command line podresources clients, but not an recommended tool (no `crictl` equivalent to say).